File: csharpcore/GildedRose/Models/Conjured.cs
using GildedRose.Abstraction;
using GildedRoseKata;

namespace GildedRose.Models
{
    public class Conjured : Item, ICustomMethod
    {
        public int SellDaysGone { get; set; }

        public void UpdateQuality()
        {
            if (this.SellDaysGone > this.SellIn && this.Quality > 1)
                this.Quality -= 2;
        }

        public void UpdateSellIn()
        {
            if (this.SellIn > 0)
                this.SellIn--;
        }
    }
}

File: Java/src/main/java/com/gildedrose/item/Conjured.java
/* (C)2022 */
package com.gildedrose.item;

public class Conjured extends Item {

    public static final String CONJURED = "Conjured";

    public Conjured(int sellIn, int quality) {
        super(CONJURED, sellIn, quality);
    }

    @Override
    public void updateQuality() {
        if (sellIn <= 0) {
            quality -= 4;
        } else {
            quality -= 2;
        }
        if (quality < 0) {
            quality = 0;
        }
    }
}
